src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,305 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
 {
-    return encoder.Encode(mICDConfigurationData->GetActiveModeDuration().count());
+    switch (request.path.mCommandId)
+    {
+    case IcdManagement::Commands::StayActiveRequest::Id: {
+// TODO(#32321): Remove #if after issue is resolved
+// Note: We only need this #if statement for platform examples that enable the ICD management server without building the sample
+// as an ICD. Since this is not spec compliant, we should remove this #if statement once we stop compiling the ICD management
+// server in those examples.
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+        Commands::StayActiveRequest::DecodableType commandData;
+        ReturnErrorOnFailure(commandData.Decode(input_arguments));
+
+        Commands::StayActiveResponse::Type response;
+        response.promisedActiveDuration = Server::GetInstance().GetICDManager().StayActiveRequest(commandData.stayActiveDuration);
+        handler->AddResponse(request.path, response);
+        return std::nullopt;
+#else
+        return Status::Success; // TODO: WHY? Is this not an error? should we return error instead?
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+    }
+    default:
+        return Status::UnsupportedCommand;
+    }
 }
 
-CHIP_ERROR ICDManagementAttributeAccess::ReadActiveModeThreshold(EndpointId endpoint, AttributeValueEncoder & encoder)
+CHIP_ERROR ICDManagementCluster::AcceptedCommands(const ConcreteClusterPath & path,
+                                                  ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder)
 {
-    return encoder.Encode(mICDConfigurationData->GetActiveModeThreshold().count());
+    BitFlags<IcdManagement::Feature> featureMap = mICDConfigurationData.GetFeatureMap();
+    const bool hasCIP                           = featureMap.Has(Feature::kCheckInProtocolSupport);
+
+    if (hasCIP)
+    {
+        static constexpr DataModel::AcceptedCommandEntry kAcceptedCommands[] = {
+            Commands::RegisterClient::kMetadataEntry,
+            Commands::UnregisterClient::kMetadataEntry,
+        };
+        ReturnErrorOnFailure(builder.ReferenceExisting(kAcceptedCommands));
+    }
+
+    if (mICDConfigurationData.GetFeatureMap().Has(Feature::kLongIdleTimeSupport) ||
+        mEnabledCommands.Has(OptionalCommands::kStayActive))
+    {
+        ReturnErrorOnFailure(builder.EnsureAppendCapacity(1));
+        ReturnErrorOnFailure(builder.Append(Commands::StayActiveRequest::kMetadataEntry));
+    }
+    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ICDManagementAttributeAccess::ReadFeatureMap(EndpointId endpoint, AttributeValueEncoder & encoder)
+CHIP_ERROR ICDManagementCluster::GeneratedCommands(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<CommandId> & builder)
 {
-    return encoder.Encode(mICDConfigurationData->GetFeatureMap());
+    BitFlags<IcdManagement::Feature> featureMap = mICDConfigurationData.GetFeatureMap();
+    const bool hasCIP                           = featureMap.Has(Feature::kCheckInProtocolSupport);
+
+    if (hasCIP)
```

Comment:
We have a separate `ICDManagementClusterWithCIP` yet the base class also has CIP checks ... why is that?

Documentation on base class seems to say that one should use CIP version when using CIP.

Reply:
Removed it